server/src/modules/notifications/agents/webhook.ts

```diff
@@ -0,0 +1,137 @@
+import axios from 'axios';
+import { get } from 'lodash';
+import { hasNotificationType } from '../notifications.service';
+import type { NotificationAgent, NotificationPayload } from './agent';
+import { SettingsService } from '../../settings/settings.service';
+import { Logger } from '@nestjs/common';
+import {
+  NotificationAgentConfig,
+  NotificationAgentKey,
+  NotificationType,
+} from '../notifications-interfaces';
+import { Notification } from '../entities/notification.entities';
+
+type KeyMapFunction = (
+  payload: NotificationPayload,
+  type: NotificationType,
+) => string;
+
+const KeyMap: Record<string, string | KeyMapFunction> = {
+  notification_type: (_payload, type) => NotificationType[type],
+  event: 'event',
+  subject: 'subject',
+  message: 'message',
+  image: 'image',
+};
+
+class WebhookAgent implements NotificationAgent {
+  public constructor(
+    private readonly appSettings: SettingsService,
+    private readonly settings: NotificationAgentConfig,
+    readonly notification: Notification,
+  ) {
+    this.notification = notification;
+  }
+
+  private readonly logger = new Logger(WebhookAgent.name);
+
+  getNotification = () => this.notification;
+
+  getSettings = () => this.settings;
+
+  getIdentifier = () => NotificationAgentKey.WEBHOOK;
+
+  private parseKeys(
+    finalPayload: Record<string, unknown>,
+    payload: NotificationPayload,
+    type: NotificationType,
+  ): Record<string, unknown> {
+    Object.keys(finalPayload).forEach((key) => {
+      if (key === '{{extra}}') {
+        finalPayload.extra = payload.extra ?? [];
+        delete finalPayload[key];
+        key = 'extra';
+      }
+
+      if (typeof finalPayload[key] === 'string') {
+        Object.keys(KeyMap).forEach((keymapKey) => {
+          const keymapValue = KeyMap[keymapKey as keyof typeof KeyMap];
+          finalPayload[key] = (finalPayload[key] as string).replace(
+            `{{${keymapKey}}}`,
+            typeof keymapValue === 'function'
+              ? keymapValue(payload, type)
+              : (get(payload, keymapValue) ?? ''),
+          );
+        });
+      } else if (finalPayload[key] && typeof finalPayload[key] === 'object') {
+        finalPayload[key] = this.parseKeys(
+          finalPayload[key] as Record<string, unknown>,
+          payload,
+          type,
+        );
+      }
+    });
+
+    return finalPayload;
+  }
+
+  private buildPayload(type: NotificationType, payload: NotificationPayload) {
+    const payloadString = Buffer.from(
+      this.getSettings().options.jsonPayload as string,
+      'base64',
+    ).toString('ascii');
+
+    const parsedJSON = JSON.parse(JSON.parse(payloadString));
```

Comment:
As you're saving the payload as JSON (unlike overseerr I guess), you don't need any of this decoding stuff.

```suggestion
    const payloadString = this.getSettings().options.jsonPayload as string;
    const parsedJSON = JSON.parse(payloadString);
```